### PR TITLE
Improved Anti-Shuffle Hug so Carrier Allies Don't Reset Jump Timers

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Xenos from the same hive as the one that drops or throws a Facehugger no longer reset that Facehugger's jump timer when they leave its tile.

## Why It's Good For The Game

Xenos from the same hive as the one that drops or throws a Facehugger no longer reset that Facehugger's jump timer when they leave its tile.

## Changelog
:cl:
code: Xenos from the same hive as the one that drops or throws a Facehugger no longer reset that Facehugger's jump timer when they leave its tile.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/facehuggers.dm [hunk 1]
```diff
@@ -39,7 +39,8 @@
 	var/leaping = FALSE
 	///What hive this hugger belongs to
 	var/hivenumber = XENO_HIVE_NORMAL
-
+	///The xeno that spawned/threw/dropped the hugger. Used for anti-shuffle
+	var/mob/living/carbon/xenomorph/source
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/facehuggers.dm:43]: No hard del protection

Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/facehuggers.dm:43]: Fixed probably

File: code/modules/mob/living/carbon/xenomorph/facehuggers.dm [hunk 2]
```diff
@@ -58,21 +59,37 @@
...
 	if(stat == CONSCIOUS)
 		lifetimer = addtimer(CALLBACK(src, .proc/check_lifecycle), FACEHUGGER_DEATH, TIMER_STOPPABLE)
 
 	if(input_hivenumber)
 		hivenumber = input_hivenumber
 
+	if(input_source)
+		facehugger_register_source(input_source)
+
+/obj/item/clothing/mask/facehugger/proc/facehugger_register_source(mob/living/carbon/xenomorph/S)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/facehuggers.dm:73]: Autodoc

File: code/modules/mob/living/carbon/xenomorph/facehuggers.dm [hunk 3]
```diff
@@ -58,21 +59,37 @@
...
+	if(source) //If we have an existing source, unregister
+		UnregisterSignal(source, COMSIG_PARENT_QDELETING)
+
+	source = S //set and register new source
+	RegisterSignal(S, COMSIG_PARENT_QDELETING, .proc/clear_hugger_source)
+
+/obj/item/clothing/mask/facehugger/proc/clear_hugger_source()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/facehuggers.dm:80]: Autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**